File: sword-means-offer/06/main.go
package main

// @Time     : 2021/6/4 13:55
// @Author   : Ranshi
// @File     : 剑指 Offer 06. 从尾到头打印链表.go

type ListNode struct {
	Val  int
	Next *ListNode
}

func reversePrint(head *ListNode) (ans []int) {
	cur := head
	for cur != nil {
		ans = append(ans, cur.Val)
		cur = cur.Next
	}
	for i, j := 0, len(ans)-1; i < j; i, j = i+1, j-1 {
		ans[i], ans[j] = ans[j], ans[i]
	}
	return
}

func main() {
}

File: sword-means-offer/06/main.py
# -*- coding: UTF-8 -*-
# @Time     : 2021/09/25 07:23
# @Author   : Ranshi
# @File     : 123.py
from typing import List

from type.list_node import ListNode


class Solution:
    def reversePrint(self, head: ListNode) -> List[int]:
        res = []
        cur = head
        while cur:
            res.append(cur.val)
            cur = cur.next
        return res[::-1]


if __name__ == "__main__":
    print(Solution().reversePrint(ListNode.init_by_lst([1, 3, 2])))
